scio-core/src/main/scala/com/spotify/scio/package.scala

```diff
@@ -78,6 +78,12 @@ package object scio {
       Await.result(self.flatMap(identity), atMost)
   }
 
+  /** Set the name of the next transform. */
+  implicit class NamedTfSCollection[T: ClassTag](val wrapped: SCollection[T]) {
+    def withName(tfName: String): SCollection[T] =
```

Comment:
I mean you might not need a implicit class, have `def .withName` on `PCollectionWrapper` would be the best/cleanest, if we can get rid of `CallSites.getCurrent` everywhere else and replace them with `this.nameProvider.get()`.